src/history.coffee

```diff
@@ -118,13 +133,18 @@ class History
         hasSeenChanges = true
     null
 
-  popChanges: (fromStack, toStack, checkpointIndex) ->
+  popChanges: (fromStack, toStack, checkpointIndex, lookBack) ->
     changes = []
-    snapshot = fromStack[checkpointIndex].snapshot
-    for entry in fromStack.splice(checkpointIndex) by -1
+    snapshotBefore = null
+    snapshotAfter = fromStack[checkpointIndex].snapshot
+    splicedEntries = fromStack.splice(checkpointIndex)
+    for entry in splicedEntries by -1
       toStack?.push(entry)
-      changes.push(entry) unless entry instanceof Checkpoint
-    {changes, snapshot}
+      if entry instanceof Checkpoint
```

Comment:
After writing this, I'm thinking maybe their names should be reversed. I was thinking of 'before' and 'after' in terms of before and after the in-progress operation is applied, but in terms of their position in the stack, it's the opposite.